src/services/book.service.ts

```diff
@@ -1,16 +1,25 @@
 import { AppDataSource } from '../config/data-source';
 import { Book } from '../entity/book.entity';
 
+const bookRepository = AppDataSource.getRepository(Book);
+
 export async function getBookCount(): Promise<number> {
-    const bookRepository = AppDataSource.getRepository(Book);
     const count = await bookRepository.count();
     return count;
 }
 
 export const getBooks = async () => {
-    const bookRepository = AppDataSource.getRepository(Book);
     return await bookRepository.find({
         order: { title: 'ASC' },
         relations: ['author']
     });
 };
+
+export const getBookDetails = async (id: number) => {
+    const book = await bookRepository.findOne({
+        relations: ['author', 'genres', 'instances'],
+        where: { id: id }
+    });
+
+    return book;
```

Comment:
Sửa tương tự bên dưới e nha
```suggestion
    return await bookRepository.findOne({
        relations: ['author', 'genres', 'instances'],
        where: { id: id }
    });
```